runtime/src/iree/base/internal/cpu.c

```diff
@@ -375,15 +375,35 @@ static void iree_cpu_initialize_from_platform_x86_64(uint64_t* out_fields) {
 
 // For now as we only need ISA feature bits and no CPU identification beyond
 // that, and as we are OK with requiring a sufficiently recent linux kernel to
-// expose the features that we need, we can just rely on the basic HWCAP way.
-#include <sys/auxv.h>
+// expose the features that we need, we can just rely on the HWPROBE way.
+// See:
+// https://github.com/torvalds/linux/blob/master/Documentation/arch/riscv/hwprobe.rst
+#include <sched.h>
+#include <sys/hwprobe.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+
+static struct riscv_hwprobe kv_pairs[] = {{RISCV_HWPROBE_KEY_IMA_EXT_0, 0}};
 
-#define IREE_HWCAP_ISA_V (1 << ('V' - 'A'))
+// NOTE: not all kernel versions have all of the cap bits we need defined so as
+// a practice we always define the feature bits we need locally.
+#define RISCV_HWPROBE_IMA_V (1 << 2)
```

Comment:
Prefix with IREE_ or #if !defined() check so we don't collide with kernels who do.